### PR TITLE
Udpate Microsoft.DevTestLab/environments to nested virtualMachines resource type

### PR DESCRIPTION
**Pull request by shichaoh**
And also update the properties schema.

### CODE REVIEW

File: schemas/2015-05-21-preview/Microsoft.DevTestLab.json [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
-    "environments": {
-      "description": "Required properties to create environment in an existing DevTest Lab",
+    "virtualMachines": {
+      "description": "Required properties to create a virtual machine in an existing DevTest Lab",
       "type": "object",
       "properties": {
         "type": {
           "enum": [
-            "Microsoft.DevTestLab/environments"
+            "Microsoft.DevTestLab/labs/virtualMachines"
```
Comment (wenwu449) [inline schemas/2015-05-21-preview/Microsoft.DevTestLab.json]: Does this resource type rename not require a apiVersion change?

File: schemas/2015-05-21-preview/Microsoft.DevTestLab.json [hunk 2]
```diff
@@ -38,38 +38,16 @@
...
-                    }
-                  }
-                },
-                { "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression" }
-              ],
-              "description": "Microsoft.DevTestLab/environments - VMs"
-            }
-          },
-          "required": [ "labId", "vms" ]
+            "customImageName": { "type": "string" },
```
Comment (wenwu449) [inline schemas/2015-05-21-preview/Microsoft.DevTestLab.json]: indent seems not correct.

File: schemas/2015-05-21-preview/Microsoft.DevTestLab.json [hunk 3]
```diff
@@ -38,38 +38,16 @@
...
+            "size": { "type": "string" },
+            "userName": { "type": "string" },
+            "password": { "type": "string" },
+            "sshKey": { "type": "string" },
+            "isAuthenticationWithSshKey": { "type": "boolean" }
```
Comment (wenwu449) [inline schemas/2015-05-21-preview/Microsoft.DevTestLab.json]: Please support expression for non-string type property.

File: schemas/2015-05-21-preview/Microsoft.DevTestLab.json [hunk 4]
```diff
@@ -38,38 +38,16 @@
...
+          }
         }
       },
-      "required": [ "type", "apiVersion", "properties", "location" ]
```
Comment (wenwu449) [inline schemas/2015-05-21-preview/Microsoft.DevTestLab.json]: location is not required for this resource?

File: schemas/2015-01-01/deploymentTemplate.json [hunk 5]
```diff
@@ -42,7 +42,7 @@
                   { "$ref": "http://schema.management.azure.com/schemas/2015-04-08/Microsoft.DocumentDB.json#/resourceDefinitions/databaseAccounts" },
                   { "$ref": "http://schema.management.azure.com/schemas/2015-06-01/Microsoft.KeyVault.json#/resourceDefinitions/secrets" },
                   { "$ref": "http://schema.management.azure.com/schemas/2015-05-21-preview/Microsoft.DevTestLab.json#/resourceDefinitions/labs" },
-                  { "$ref": "http://schema.management.azure.com/schemas/2015-05-21-preview/Microsoft.DevTestLab.json#/resourceDefinitions/environments" },
```
Comment (wenwu449) [inline schemas/2015-01-01/deploymentTemplate.json]: Please update 2014-04-01-preview/deploymentTemplate.json too.

File: schemas/2015-05-21-preview/Microsoft.DevTestLab.json (hunk 1 above)
Comment (shichaoh) [inline schemas/2015-05-21-preview/Microsoft.DevTestLab.json]: We are still in preview. We are helping customers to migrate to the new schema.

File: schemas/2015-05-21-preview/Microsoft.DevTestLab.json (hunk 4 above)
Comment (shichaoh) [inline schemas/2015-05-21-preview/Microsoft.DevTestLab.json]: In our business logic. location is not required for virtualMachine. The location will be the same as the parent resource (lab). If ARM requires it, we should add it back.

**Comment by shichaoh** [comment]
We have a couple of more changes and created a new PR here:
https://github.com/Azure/azure-resource-manager-schemas/pull/131

I will abandon this PR.

**Pull request closed without merging**